gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -23,6 +23,37 @@ from .transports.base import {{ service.name }}Transport
 from .transports.grpc import {{ service.name }}GrpcTransport
 
 
+def _get_default_mtls_endpoint(api_endpoint):
+    """Convert api endpoint to mTLS endpoint.
+    Convert "*.sandbox.googleapis.com" and "*.googleapis.com" to
+    "*.mtls.sandbox.googleapis.com" and "*.mtls.googleapis.com" respectively.
+    Args:
+        api_endpoint (Optional[str]): the api endpoint to convert.
+    Returns:
+        str: converted mTLS api endpoint.
+    """
+    if (
+        api_endpoint is None
+        or api_endpoint.find("mtls.sandbox.googleapis.com") != -1
+        or api_endpoint.find("mtls.googleapis.com") != -1
+        or api_endpoint.find(".googleapis.com") == -1
+    ):
```

Comment:
What do you think about moving this logic into regular expressions? Something like:
```python

# Class attribute
MTLS_ENDPOINT_RE = re.compile(r"(?P<name>[^.]+)(?P<mtls>\.mtls)?(?P<sandbox>\.sandbox)?(?P<googledomain>\.googleapis\.com)?")

...
if not api_endpoint:
    return api_endpoint

m = self.MTLS_ENDPOINT_RE.match(api_endpoint)
name, mtls, sandbox, googledomain = m.groups()
if mtls or not googledomain:
    return api_endpoint

if sandbox:
    return re.sub('sandbox.googleapis.com', 'mtls.sandbox.googleapis.com', api_endpoint)

return re.sub('.googleapis.com', '.mtls.googleapis.com')
```

I could go either way; the one part I think is clearer is removing the big expression in the `if` statement.

Reply:
yes, updated the code with the regex. Thank you!